language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/BallerinaDocumentServiceImpl.java

```diff
@@ -149,14 +150,11 @@ public BallerinaDocumentServiceImpl(WorkspaceManager workspaceManager, LanguageS
                 if (filePath.isEmpty()) {
                     return ballerinaProject;
                 }
-                Optional<Project> project = this.workspaceManager.project(filePath.get());
-                if (project.isEmpty()) {
-                    return ballerinaProject;
-                }
-                ballerinaProject.setPath(project.get().sourceRoot().toString());
-                ProjectKind projectKind = project.get().kind();
+                Project project = ProjectLoader.loadProject(filePath.get());
+                ballerinaProject.setPath(project.sourceRoot().toString());
+                ProjectKind projectKind = project.kind();
```

Comment:
Why do we use ProjectLoader here? Can't we use the workspace manager to access the project, given the filePath?

Reply:
VSCode has a popup message upon file open, which calls this API before the file is actually open in the editor. The workspace manager doesn't contain the required info at that time. We have to newly load the project to access the info.

Reply:
Cool, +1 to proceed as such

Reply:
Can't we get these details through `ProjectPaths.isStandaloneBalFile(path)` and `ProjectPaths.packageRoot(path)` util methods? `ProjectLoader.loadProject(filePath.get())` is a costly operation.

Reply:
In order to get the package name then we have to do some path operations right? If there are any package structure change wouldn't it be an issue?

Reply:
Hmm yeah, that can cause problems in future. This is a valid use-case`workspace.project()` api,  for how about extending functionality of `workspace.project()` functionality to load a project if not loaded. Reason is here we load a new project instance and discard it.